packages/wallet-core/src/types.ts

```diff
@@ -93,15 +93,13 @@ type _Objective<Name, Data> = {
   type: Name;
   data: Data;
 };
-export type OpenChannel = _Objective<
-  'OpenChannel',
-  {
-    targetChannelId: string;
-    fundingStrategy: FundingStrategy;
-    role?: 'app' | 'ledger'; // Default should be app
-    fundingLedgerChannelId?: string;
-  }
->;
+
+type OpenChannelFunding =
+  | {fundingStrategy: 'Direct' | 'Fake' | 'Unknown'; role?: 'app' | 'ledger'} // Default should be 'app'
+  | {fundingStrategy: 'Ledger'; fundingLedgerChannelId: string}
+  | {fundingStrategy: 'Virtual'; rest: unknown}; // TODO: Updated in a later PR
+
+export type OpenChannel = _Objective<'OpenChannel', {targetChannelId: string} & OpenChannelFunding>;
```

Comment:
This seems like a good move to me. Some questions:

1. In the 'Ledger' branch of the union type, you made `fundingLedgerChannelId` a required field. Should we do the same with the `role` field in the `Direct` branch? 
2. Is `rest` just a placeholder for the "rest" of the parameters? If so, maybe go with `additionalParameters` or something like that?
3. You are no longer using the `FundingStrategy` type, still imported in this file. It seems like that is requiring some type assertions elsewhere in this changeset. What are your thoughts on tidying that up?

Reply:
1. Good question. My interpretation of the "Default should be app" comment before was that in a directly funded channel, `role === undefined` implies it is an `'app'` channel, and I didn't intend to change that.

I think this implication is not good, as there are then two ways that a channel can be marked as an app channel: `role === undefined` and `role === 'app'`. I am happy to make it a required property.

2. Yes. I might remove `rest` entirely, and leave `Virtual` as another unsupported type, for the purpose of this PR.

3. Thanks for the pointer. I'll look into it.

Reply:
For (3), do you mean the type assertion [here](https://github.com/statechannels/statechannels/pull/3445/files#diff-4db1091066d3e5e5327bcb48b266dbe47373bac9ff8507c540cab22ed69c8f66R104-R106)?

I am not really sure how to tidy it up.

Reply:
Yes that's the one. I don't know how, either 😛